lib/msfdb_helpers/pg_ctl.rb

```diff
@@ -28,6 +40,37 @@ def init(msf_pass, msftest_pass)
       restart
     end
 
+    # Creates and attempts to execute a testfile in the specified directory,
+    # to determine if it is mounted with NOEXEC flags.
+    def test_executable_file(path)
+      begin
+        file_name = File.join(path, 'msfdb_testfile')
+        File.open(file_name, 'w') do |f|
+          f.puts "#!/bin/bash\necho exec"
+        end
+        File.chmod(0744, file_name)
+        
+        if run_cmd(file_name)
+          File.open("#{@db}/postgresql.conf", 'a') do |f|
+            f.puts "unix_socket_directories = \'#{path}\'"
+          end
+          puts "Creating db socket file at #{path}"
```

Comment:
Should this not be at the start of the function? Putting it at the end here seems odd given that you don't actually do anything after this is printed out, all the work here is done before this gets printed out.

Reply:
Hmm so seems I misunderstood what was going on here, and this is actually fine. Marking as resolved.